### PR TITLE
refactor: Move remaining methods from UsersResponder to UserService and UserRestService

### PR DESCRIPTION
**Pull request by seakayone**
- **Move add groups methods from UsersResponder to RestService**
- **Move validation business logic from RestService to UserService**
- **Merge UserResponseADM with UserOperationResponseADM**
- **Move changePassword to UsersRestService and UserService**
- **Move changeSystemAdmin to UsersRestService and UserService**
- **rename private functions**
- **Move delete and change status to UsersRestService and UserService**
- **Move update user basic info to UsersRestService and UserService**
- **Move create new User to UsersRestService and UserService**
- **Move remaining methods to UsersRestService and UserService**

I will keep `UsersResponder` for now and will resolve the circular dependency in a followup (the UserService needs GroupResponderADM and vice versa).

### Task Description/Number

Issue Number: DEV-

### PR Type

- [ ] build/chore: maintenance tasks (no production code change)
- [ ] docs: documentation changes (no production code change)
- [ ] feat: represents new features
- [ ] fix: represents bug fixes
- [ ] perf: performance improvements
- [x] refactor: represents production code refactoring
- [ ] test: adding or refactoring tests (no production code change)

### Basic requirements for bug fixes and features

- [ ] Tests for the changes have been added
- [ ] Docs have been added / updated

### Does this PR introduce a breaking change?

- [ ] Yes

### Does this PR change client-test-data?

- [ ] Yes

### CODE REVIEW

**Review by BalduinLandolt — APPROVED** [review]
LGTM

Seems there are file headers missing

File: webapi/src/main/scala/org/knora/webapi/responders/IriService.scala [hunk 1]
```diff
@@ -58,6 +61,15 @@ final case class IriService(
   def isClassUsedInData(classIri: SmartIri): Task[Boolean] =
     triplestore.query(Select(sparql.v2.txt.isClassUsedInData(classIri))).map(_.results.bindings.nonEmpty)
 
+  def checkOrCreateNewUserIri(entityIri: Option[UserIri]): Task[UserIri] =
+    for {
+      // check the custom IRI; if not given, create an unused IRI
+      customUserIri <- ZIO.foreach(entityIri.map(_.value))(iriConverter.asSmartIri)
+      userIri <-
+        checkOrCreateEntityIri(customUserIri, UserIri.makeNew.value)
+          .flatMap(iri => ZIO.fromEither(UserIri.from(iri)).orElseFail(BadRequestException(s"Invalid User IRI: $iri")))
```
[the remark this answers is not in the document]
Comment (seakayone) [reply webapi/src/main/scala/org/knora/webapi/responders/IriService.scala]: => f133e6f1a07dada92f1704ed44b47b8d8e111674

**Comment by seakayone** [comment]
> LGTM
> 
> Seems there are file headers missing

=> fc3fe3d89e176c3bff5c20c755f0f3e1d4d92134

**Review by mpro7 — APPROVED** [review]
Great, thanks!

**Review by siers — APPROVED** [review]

**Pull request merged**